.github/workflows/ci.yml

```diff
@@ -89,3 +89,41 @@ jobs:
                   name: Screenshots-${{ matrix.tag }}-${{ matrix.browser }}
                   path: screenshots/
                   if-no-files-found: error
+
+    test-package:
+
+        needs: [pre-commit]
+
+        strategy:
+            matrix:
+                tag: [latest]
+                python-version: ['3.8', '3.9', '3.10']
+            fail-fast: false
+
+        runs-on: ubuntu-latest
+        timeout-minutes: 30
+
+        steps:
+
+            - name: Check out app
+              uses: actions/checkout@v3
```

Comment:
This is sort of general question. The `pre-commit` action was executed before and this step was there too. SInce `test-package` depends on `pre-commit`, why do we rerun this step then?

If we redo it anyways, what is the point of `needs`?

Reply:
From the doc https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds, what I understand is it only requires the job to be passed but not inherit the state/environment of the job. The jobs are still independent of the environment.